File: Heaps/deletGreatestValueInEachRow/pq.py
import heapq

class Solution:
    def deleteGreatestValue(self, grid: list[list[int]]) -> int:
        heaps = [] # max heap for each row
        for row in grid:
            heap = [-val for val in row] # max heap 
            heapq.heapify(heap)
            heaps.append(heap)
        result = 0
        n = len(grid[0])
        for col in range(n):
            curMax = 0
            for heap in heaps:
                val = -heapq.heappop(heap)
                curMax = max(curMax,val)
            result += curMax
        return result

File: Heaps/deletGreatestValueInEachRow/pq.cpp
#include <queue>
#include <vector>
using namespace std;
class Solution {
public:
    int deleteGreatestValue(vector<vector<int>>& grid) {
        int m = grid.size(), n = grid[0].size();
        
        vector<priority_queue<int>> heaps(m); // max_heap by def
        for (int i = 0; i < m; i++)
            for (int val : grid[i])
                heaps[i].push(val);
        
        int result = 0;
        
        for (int col = 0; col < n; col++) {
            int roundMax = 0;
            for (int i = 0; i < m; i++) {
                roundMax = max(roundMax, heaps[i].top());
                heaps[i].pop();
            }
            result += roundMax;
        }
        
        return result;
    }
};